File: network/src/main/java/com/naggaro/newsapp/network/RequestInterceptor.java
package com.naggaro.newsapp.network

import com.seed.roundrobin.views.system.RequestHeaders;
import okhttp3.Headers;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

import java.io.IOException;

/**
 */

public class RequestInterceptor implements Interceptor {
    private RequestHeaders requestHeaders;
    public RequestInterceptor(RequestHeaders requestHeaders) {
        this.requestHeaders = requestHeaders;
    }
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request original = chain.request();
        Request.Builder builder = original
                .newBuilder()
                .headers(Headers.of(requestHeaders.getHeaders()))
                .method(original.method(), original.body());
        Request newRequest = builder.build();
        return chain.proceed(newRequest);
    }


}

File: dtos/src/main/java/com/naggaro/dtos/network/RequestInterceptor.kt
package com.naggaro.dtos.network

import okhttp3.Headers
import okhttp3.Headers.Companion.toHeaders
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import java.io.IOException

/**
 */
class RequestInterceptor(private val requestHeaders: RequestHeaders) : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val original: Request = chain.request()
        val builder = original
            .newBuilder()
            .headers(requestHeaders.headers.toHeaders())
            .method(original.method, original.body)
        val newRequest: Request = builder.build()
        return chain.proceed(newRequest)
    }
}